HW0/P4/P4.py

```diff
@@ -65,12 +86,48 @@
 
     # B = ?
     # C = ?
+    A_matrix = propagation
+    B_diag = np.array([bx, by, bz, bvx, bvy, bvz])
+    B_matrix = np.diag(B_diag)
+    C_matrix = np.hstack([np.diag(np.array([rx, ry, rz])), np.zeros([3,3])])
 
     # Initial conditions for s0 and Sigma0
     # Compute the rest of sk using Eqs (2), (3), (4), and (5)
+    s2_matrix = np.zeros([6, K])
+    s2_matrix[:, 0] = np.array([0.0, 0.0, 2.0, 15.0, 3.5, 4.0]).transpose()
+    sig_diag = np.array([1, 1, 1, 1, 1, 1]) * 0.01
+    sig = np.diag(sig_diag)
+
+    def predictS():
+        s2_est = np.add(np.dot(propagation, s2_prev), acceleration)
+        return s2_est
+
+    def predictSig():
+        sig_est = np.linalg.inv(((A_matrix * sig) * A_matrix.transpose()) + B_matrix * B_matrix.transpose())
+        return sig_est
+
+    def updateSig():
+        sig = np.linalg.inv(sig_est + np.dot(C_matrix.transpose(), C_matrix))
+        return sig
+
+    def updateS():
+        s2 = np.dot(sig, (np.dot(sig_est, s2_est) + np.dot(C_matrix.transpose(), s_measure_trans[:, k])))
+        return s2
```

Comment:
As a suggestion, I think its important for functions to be independent just because it allows you portability first of all and secondly it allows you to test them irrespective of the case. I know it wasn't exactly a requirement in this case but this is something I have previously been suggested to do and thought it might just help you as well.